src/Microsoft.Health.SqlServer/DefaultSqlConnectionBuilder.cs

```diff
@@ -82,6 +82,23 @@ public async Task<SqlConnection> GetReadOnlySqlConnectionAsync(string initialCat
         return new SqlConnection(builder.ToString()) { RetryLogicProvider = _retryProvider };
     }
 
+    [SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Caller must dispose result.")]
+    public async Task<SqlConnection> GetSqlConnectionAsync(bool isReadOnly, string applicationName, CancellationToken cancellationToken)
```

Comment:
This method will be used in FHIR for read only connections and connections required to have application name set.